src/fides/api/api/v1/endpoints/privacy_preference_endpoints.py

```diff
@@ -231,12 +277,14 @@ def _get_request_origin_and_config(
     return origin, experience_config_history_id
 
 
-def supplement_privacy_preferences_with_user_and_experience_details(
-    db: Session, request: Request, data: PrivacyPreferencesRequest
-) -> PrivacyPreferencesCreate:
+def supplement_request_with_user_and_experience_details(
+    db: Session,
+    request: Request,
+    data: Union[PrivacyPreferencesRequest, NoticesServedRequest],
+    resource_type: Union[Type[PrivacyPreferencesCreate], Type[NoticesServedCreate]],
+) -> Union[PrivacyPreferencesCreate, NoticesServedCreate]:
```

Comment:
Generalizing here to share when saving notices and when notices were served. Both tables track a fair amount of details about users for consent reporting